contrib/platform/src/com/sun/jna/platform/win32/Psapi.java

```diff
@@ -320,4 +338,79 @@ class PERFORMANCE_INFORMATION extends Structure {
         public DWORD ProcessCount;
         public DWORD ThreadCount;
     }
+
+    @FieldOrder({"Flags"})
+    class PSAPI_WORKING_SET_EX_BLOCK extends Structure {
+        public ULONG_PTR[2] Flags = new ULONG_PTR[Native.POINTER_SIZE == 8 ? 1 : 2];
+        private Long innerValue;
+        private void setInnerValue() {
+            if (innerValue != null) {
+                return;
+            } else if (this.Flags.length == 2) {
+                Long upper = Optional.ofNullable(Flags[1]).map((elem) -> elem.longValue()).orElse(0l);
+                upper = upper << 32;
+                int lower = Optional.ofNullable(Flags[0]).map((elem) -> elem.longValue()).orElse(0l);
+                this.innerValue = upper | lower;
+            } else {
+                this.innerValue = Optional.ofNullable(Flags[0]).map((elem) -> elem.longValue()).orElse(0l);
+            }
+        }
+
+        public Boolean Valid() {
+            setInnerValue();
+            return innerValue & 1 == 1;
+        }
+
+        public Integer ShareCount() {
+            setInnerValue();
+            return Math.toIntExact(innerValue >> 1 & 0b111);
```

Comment:
I think if you mask before shifting you can avoid the `toIntExact`.   You might even write a helper method something like this (untested):
```java
private int getBitFieldValue(int lowBit, int highBit) {
  long bitmask = 0;
  for (int b = lowBit; b<= highBit; b++) {
    bitMask |= 1 << b;    
  }
  return (innerValue & bitMask) >> lowBit;
}
```
Then you could just make this method:
```java
public int getShareCount() {
   return getBitFieldValue(1, 3);
}
```

Reply:
So I realized `toIntExact()` is just a cast which I neglected to include in my helper function, and I don't think the mask vs. shift order matters.  But I do like the helper function idea.   Also you don't need to write methods to return the "reserved" bits.